### PR TITLE
Feature/quotes-refresh

### PR DESCRIPTION
**Pull request by alaahd**
**Description**

* [x] Adding data polling for fetching new quotes using time interval. 
* [x] Updating the header property in ScreenLayout component to accept React FC in addition to string type.
* [x] Updating story book.

**Checklist**
* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

File: app/components/UI/FiatOnRampAggregator/sdk/index.tsx [hunk 1]
```diff
@@ -125,10 +125,12 @@ export function useSDKMethod<T extends keyof IOnRampSdk>(
...
-			// @ts-expect-error spreading params error
-			const sdkMethod = (...a) => sdk[method](...a);
-			const response = await sdkMethod(...params);
-			setData(response);
+			if (sdk) {
+				// @ts-expect-error spreading params error
+				const sdkMethod = (...a) => sdk[method](...a);
+				const response = await sdkMethod(...params);
+				setData(response);
+			}
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/sdk/index.tsx:133]: How frequent does this case happens? Should we log this?

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js [hunk 2]
```diff
@@ -1,24 +1,92 @@
...
+import { ScrollView, TouchableOpacity } from 'react-native-gesture-handler';
+import { colors } from '../../../../styles/common';
+import useInterval from '../../../hooks/useInterval';
+import ScreenView from '../../FiatOrders/components/ScreenView';
+import StyledButton from '../../StyledButton';
+import Device from '../../../../util/device';
+
+const POLLING_INTERVAL = 15000;
+const POLLING_INTERVAL_HIGHLIGHT = 10000;
+const POLLING_CYCLES = 2;
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js:20]: Can you move this to the AppContants, read them on the SDKProvider from AppContants and provide them through the provider object (along with `sdk` and selected state)?

This in the future might be a setting coming from the SDK/Server

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js [hunk 3]
```diff
@@ -71,34 +234,44 @@ const GetQuotes = () => {
...
+							<Text centered>{strings('fiat_on_ramp_aggregator.loading')}</Text>
+						) : !quotes.length ? (
+							<Text black center>
+								No providers available!
+							</Text>
+						) : (
+							quotes
+								.filter(
+									({ error, errorCode, amountIn, amountOut }) =>
+										!error && !errorCode && amountIn && amountOut
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js:254]: Why is `error` not enough?

File: app/components/UI/FiatOnRampAggregator/sdk/index.tsx (hunk 1 above)
Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/sdk/index.tsx]: It is very intermittent. it happens mostly with a fresh start of the application.

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js (hunk 3 above)
Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js:263]: it is enough, but this is a more religious check!

File: app/components/UI/FiatOnRampAggregator/sdk/index.tsx (hunk 1 above)
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/sdk/index.tsx]: We will remove this provider from the app startup (splitting orders/buy flow), however we should definitely keep an eye on this when the buy flow is mounted.

Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/sdk/index.tsx]: Indeed but we are going to point to another provider which could still need a reference to another sdk instance required for order management!

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js (hunk 3 above)
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js:263]: Let's just use error please 🙏

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js [hunk 4]
```diff
@@ -16,27 +23,85 @@ const styles = StyleSheet.create({
 	row: {
 		marginVertical: 8,
 	},
+	firstRow: {
+		marginTop: 0,
+		marginBottom: 8,
+	},
+	timerWrapper: {
+		backgroundColor: colors.grey000,
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js:31]: use colors from theme, search for `createStyles` in the codebase for reference

**Review by wachunei — CHANGES_REQUESTED** [review]

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js (hunk 4 above)
Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js:31]: we are already using colors from theme

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js [hunk 5]
```diff
@@ -19,63 +18,64 @@ import { useTheme } from '../../../../util/theme';
...
+			fontSize: 14,
+		},
+		errorIcon: {
+			fontSize: 46,
+			marginVertical: 4,
+			color: colors.red,
+		},
+		expiredIcon: {
+			color: colors.blue,
+		},
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js:68]: These colors are not present in the design tokens, I think it should be `success` and `error` I'm not sure, please reach out to design team 🙏 https://github.com/MetaMask/design-tokens/blob/main/src/colors/index.ts

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js (hunk 4 above)
Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js]: Fixed

Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js]: Fixed.

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js (hunk 5 above)
Comment (alaahd) [reply app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js]: Fixed

**Review by wachunei — APPROVED** [review]

**Pull request merged**